scripts/monitor/visualization/main.py

```diff
@@ -0,0 +1,43 @@
+#!
+import pickle
+
+from template import *
+from week_one import *
+from week_two import *
+
+ds_meta_runtime = pickle.load(open("monitoring_dataset_50e68afe-a342-4357-9397-4de7b9300b10.metadata_runtime.pickle", "rb"))
+ds_metrics = pickle.load(open("monitoring_dataset_50e68afe-a342-4357-9397-4de7b9300b10.metrics.pickle", "rb"))
+# above is very temporary measure
+
+
+def generate_individual_plots():
+    pass
+
+
+def generate_report_homepage(meta_runtime: pd.DataFrame, metrics: pd.DataFrame):
+
+    html_template = HTML_STYLE+HTML_HEADING+generate_html_tables_section(table_html(meta_runtime, metrics))+HTML_VIOLIN+HTML_SCATTER
+    with open('home.html', 'w') as home:
+        #home.write('') #alternate
+        home.write(html_template)
+
+
+def table_html(meta_runtime: pd.DataFrame, metrics: pd.DataFrame):
+    dicts = num_task_shards(meta_runtime)
+    shard_count_table = pd.DataFrame(data=zip(dicts.keys(), dicts.values()), columns=["Task", "Shard Count"]).to_html(justify="center")
+    shardless_table = shardless_task_table(meta_runtime, metrics).to_html(justify="center")
+    return [shardless_table.replace("<td", '''<td style="text-align:center"'''), shard_count_table.replace("<td", '''<td style="text-align:center"''')]
+
+
+def main(meta_runtime: pd.DataFrame, metrics: pd.DataFrame):
```

Comment:
nice!
Now one improvements, is to have a data loading function that loads pickle files, and other functions can use the returned dataframes.
This means lines 8-9 isn't necessary any more. They will be the data loading function.